### PR TITLE
Simplify "bool x = y ? true : false". Remove unused function and trailing semicolon.

### PR DESCRIPTION
**Pull request by practicalswift**
Three minor cleanups:
* Simplify `bool x = y ? true : false` to `bool x = y`
* Remove unused function `InsecureRandBytes(size_t len)`
* Remove accidental trailing semicolon

### CODE REVIEW

**Review by dcousens — APPROVED** [review]

**Comment by achow101** [comment]
utACK a65124bc44515c60e13e74ec83f6adbb8cc48a68

**Comment by paveljanik** [comment]
utACK a65124b

**Comment by fanquake** [comment]
utACK a65124b

File: src/rpc/misc.cpp [hunk 1]
```diff
@@ -212,8 +212,8 @@ UniValue validateaddress(const JSONRPCRequest& request)
 
 #ifdef ENABLE_WALLET
         isminetype mine = pwallet ? IsMine(*pwallet, dest) : ISMINE_NO;
-        ret.push_back(Pair("ismine", (mine & ISMINE_SPENDABLE) ? true : false));
-        ret.push_back(Pair("iswatchonly", (mine & ISMINE_WATCH_ONLY) ? true: false));
+        ret.push_back(Pair("ismine", static_cast<bool>(mine & ISMINE_SPENDABLE)));
```
Comment (laanwj) [inline src/rpc/misc.cpp]: Wouldn't simply `bool(mine & ISMINE_SPENDABLE)` work?

**Comment by practicalswift** [comment]
@laanwj Good point! Now using the suggested `bool(mine & ISMINE_SPENDABLE)` form.

**Comment by sipa** [comment]
utACK 67ca816849d0aa292a138e6c08edafd17ad6a948

**Pull request merged**